Assets/Scripts/HouseDrawer.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Road;
+using Zenject;
+
+public class HouseDrawer : MonoBehaviour{
+
+	[Inject]
+	BoardTranslator translator;
+
+	[Inject]
+	Installer.Settings.RoadTiles tiles;
+
+	public List<GameObject> DrawHouses (RoadSegment[] roadSegments, HashSet<Coordinate> destinationCoords) {
+		List<GameObject> houses = new List<GameObject>();
+		foreach (RoadSegment roadSegment in roadSegments) {
+			if (destinationCoords.Contains (roadSegment.coords)) {
+				houses.Add(DrawHouseAt(roadSegment));
+			}
+		}
+		return houses;
+	}
+
+	private GameObject DrawHouseAt (RoadSegment roadSegment) {
+		switch (roadSegment.roadType) {
+		case RoadType.DeadEnd:
+			return DrawHouseAtDeadEnd(roadSegment.coords, roadSegment.direction);
+		case RoadType.Straight:
+			return DrawHouseAtStraight(roadSegment.coords, roadSegment.direction);
+		case RoadType.Turn:
+			return DrawHouseAtTurn(roadSegment.coords, roadSegment.direction);
+		case RoadType.TJunction:
+			return DrawHouseAtTJunction(roadSegment.coords, roadSegment.direction);
+		default:
+			return null; // Shouldn't get here - houses at crossroads not supported
+		}
+	}
+
+	private GameObject DrawHouseAtDeadEnd (Coordinate coords, Direction direction) {
+		float sceneX = translator.translateToSceneRow(coords.x);
+		float sceneY = translator.translateToSceneColumn(coords.y);
+
+		Vector3 roadPosition = new Vector3(sceneX, sceneY, 0f);
+
+		return Instantiate(tiles.houseTile, roadPosition + (ToDirectionVector(ToRadians((float)direction)) * 0.75f),
+				Quaternion.Euler(0, 0, (float)direction + 90)) as GameObject;
+	}
+	
+	private GameObject DrawHouseAtStraight (Coordinate coords, Direction direction) {
+		float sceneX = translator.translateToSceneRow(coords.x);
+		float sceneY = translator.translateToSceneColumn(coords.y);
+
+		Vector3 roadPosition = new Vector3(sceneX, sceneY, 0f);
+
+		return Instantiate(tiles.houseTile, roadPosition - (ToDirectionVector(ToRadians((float)direction)) * 0.75f),
```

Comment:
is there any way to use Euler angles for this? or at least have the constants in Euler angles (i.e. in degrees not radians)

Reply:
referring to the 0.75f

Reply:
The 0.75 is the distance, not the angle itself (if I remember correctly). This was how I tweaked the houses further away from the road when the image was changed.

Here we're taking the direction and using this to make a vector of length 1 in that direction. We multiply this by the number, in this case 0.75f, to give us a vector of length 0.75f in that direction.

Taking this from the road position allows us to position the house relatively to the road.

I'm 90% sure there should be a nice unity way to do this, but I didn't know at the time. I'd be interested in trying to find that afterwards.

About the use of radians rather than degrees. This was just because of the sin and cos functions I used in the ToDirectionVector method. They required radians.